File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
# 아이디어: sum hash 활용
# 알고리즘: hash -> O(N)

def solution(participant, completion):
    hash_dict = dict()
    sum_value = 0
    
    for p in participant:
        hash_dict[hash(p)] = p
        sum_value += hash(p)
    
    for c in completion:
        sum_value -= hash(c)
    
    return hash_dict[sum_value]

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
// 아이디어: 정렬
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Arrays.sort(participant);
        Arrays.sort(completion);
        
        for (int i = 0; i < completion.length; i++) {
            // 현재 명단이 다르다면
            if (!participant[i].equals(completion[i])) {
                return participant[i];
            } 
        }
        
        return participant[participant.length - 1];
    }
}